spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/RdbmsTaskDefinitionRepository.java

```diff
@@ -0,0 +1,297 @@
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * RDBMS implementation of {@link TaskDefinitionRepository}.
+ *
+ * @author Glenn Renfro
+ */
+public class RdbmsTaskDefinitionRepository implements TaskDefinitionRepository {
+
+	public static final String DEFAULT_TABLE_PREFIX = "TASK_";
+
+	public static final String SELECT_CLAUSE = "DEFINITION_NAME, DEFINITION ";
+
+	public static final String FROM_CLAUSE = "%PREFIX%DEFINITION ";
+
+	public static final String LIST_OF_NAMES = "listnames";
+
+	public static final String TASK_DEFINITION_NAME_WHERE_CLAUSE = "where DEFINITION_NAME = ? ";
+
+	public static final String TASK_DEFINITION_NAME_IN_CLAUSE = "where DEFINITION_NAME in ( :" + LIST_OF_NAMES + ") ";
+
+	private static final String FIND_ALL_DEFINITIONS = "SELECT " + SELECT_CLAUSE + "FROM "
+			+ FROM_CLAUSE;
+
+	private static final String FIND_DEFINITIONS_WHERE_IN_CLAUSE = FIND_ALL_DEFINITIONS
+			+ TASK_DEFINITION_NAME_IN_CLAUSE;
+
+	private static final String SAVE_TASK_DEFINITION = "INSERT into %PREFIX%DEFINITION"
+			+ "(DEFINITION_NAME, DEFINITION)"
+			+ "values (?, ?)";
+
+	private static final String TASK_DEFINITION_COUNT = "SELECT COUNT(*) FROM "
+			+ FROM_CLAUSE;
+
+	private static final String TASK_DEFINITION_COUNT_BY_NAME = "SELECT COUNT(*) FROM " + FROM_CLAUSE
+			+ TASK_DEFINITION_NAME_WHERE_CLAUSE;
+
+	private static final String FIND_TASK_DEFINITION_BY_NAME = FIND_ALL_DEFINITIONS + TASK_DEFINITION_NAME_WHERE_CLAUSE;
+
+	private static final String DELETE_FROM_TASK_DEFINITION = "DELETE FROM " + FROM_CLAUSE;
+
+	private static final String DELETE_FROM_TASK_DEFINITION_BY_NAME = DELETE_FROM_TASK_DEFINITION
+			+ TASK_DEFINITION_NAME_WHERE_CLAUSE;
+
+	private String tablePrefix = DEFAULT_TABLE_PREFIX;
+
+	private JdbcOperations jdbcTemplate;
+
+	private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+	private DataSource dataSource;
+
+	private Map<String, Order> orderMap;
+
+	public RdbmsTaskDefinitionRepository(DataSource dataSource) {
+		Assert.notNull(dataSource);
+		this.jdbcTemplate = new JdbcTemplate(dataSource);
+		this.namedParameterJdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+		this.dataSource = dataSource;
+		orderMap = new TreeMap<>();
+		orderMap.put("DEFINITION_NAME", Order.ASCENDING);
+	}
+
+	@Override
+	public Iterable<TaskDefinition> findAll(Sort sort) {
+		Assert.notNull(sort, "sort must not be null");
+		Iterator<Sort.Order> iter = sort.iterator();
+		String query = FIND_ALL_DEFINITIONS  + "ORDER BY " ;
+
+		while (iter.hasNext()) {
+			Sort.Order order = iter.next();
+			query = query + order.getProperty() + " " + order.getDirection();
+			if (iter.hasNext()) {
+				query = query + ", ";
+			}
+		}
+		return Collections.unmodifiableCollection(
+				jdbcTemplate.query(prefixSql(query), new TaskDefinitionRowMapper()));
```

Comment:
Why the unmodifiable collection here?  We need it in the in memory option because we wanted to control who updated the underlying datastore.  Here that isn't an issue.